Framework/API/src/AnalysisDataService.cpp

```diff
@@ -73,16 +73,18 @@ AnalysisDataServiceImpl::isValid(const std::string &name) const {
 void AnalysisDataServiceImpl::add(
     const std::string &name,
     const boost::shared_ptr<API::Workspace> &workspace) {
-  verifyName(name);
+  auto group = boost::dynamic_pointer_cast<WorkspaceGroup>(workspace);
+  verifyName(name, group);
+
   // Attach the name to the workspace
   if (workspace)
     workspace->setName(name);
   Kernel::DataService<API::Workspace>::add(name, workspace);
 
   // if a group is added add its members as well
-  auto group = boost::dynamic_pointer_cast<WorkspaceGroup>(workspace);
   if (!group)
```

Comment:
Should we move this check up with `auto group = ....`

Reply:
No, because we still want to be able to add normal workspaces to the ADS